sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/MappingsBuilder.java

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.monitor.opentelemetry.exporter.implementation;
+
+import com.azure.monitor.opentelemetry.exporter.implementation.builders.AbstractTelemetryBuilder;
+import com.azure.monitor.opentelemetry.exporter.implementation.utils.Trie;
+import io.opentelemetry.api.common.AttributeKey;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.Arrays.asList;
+
+class MappingsBuilder {
+
+    // TODO need to keep this list in sync as new semantic conventions are defined
+    private static final Set<String> STANDARD_ATTRIBUTE_PREFIXES =
+        new HashSet<>(
+            asList(
+                "http.",
+                "db.",
+                "message.",
+                "messaging.",
+                "rpc.",
+                "enduser.",
+                "net.",
+                "peer.",
+                "exception.",
+                "thread.",
+                "faas.",
+                "code.",
+                "job.", // proposed semantic convention which we use for job,
+                "applicationinsights.internal."));
+
+    private final Map<String, ExactMapping> exactMappings = new HashMap<>();
+    private final Trie.Builder<PrefixMapping> prefixMappings = Trie.newBuilder();
+
+    MappingsBuilder() {
+        // ignore all standard attribute prefixes
+        for (String prefix : STANDARD_ATTRIBUTE_PREFIXES) {
+            prefixMappings.put(prefix, (telemetryBuilder, key, value) -> {
+            });
+        }
+    }
+
+    MappingsBuilder ignoreExact(String key) {
+        exactMappings.put(key, (telemetryBuilder, value) -> {
+        });
+        return this;
+    }
+
+    MappingsBuilder ignorePrefix(String prefix) {
+        prefixMappings.put(prefix, (telemetryBuilder, key, value) -> {
+        });
+        return this;
+    }
+
+    MappingsBuilder exact(String key, ExactMapping mapping) {
+        exactMappings.put(key, mapping);
+        return this;
+    }
+
+    MappingsBuilder prefix(String prefix, PrefixMapping mapping) {
+        prefixMappings.put(prefix, mapping);
+        return this;
+    }
+
+    MappingsBuilder exactString(AttributeKey<String> attributeKey, String propertyName) {
```

Comment:
If all these methods using a lambda are called often, it may have a significant overhead for the heap allocation.

Reply:
there are only called at startup to construct the 3 Mappings objects